lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/javyenware/CsvDataLoader.java

```diff
@@ -85,12 +85,21 @@ private void populateGuestsWithCheckouts(String filePath, List<LibraryGuestModel
         // Map<String, List<CheckoutModel>> checkoutsByGuestEmail = new HashMap<>(); starts a new hashmap (a hashmap is data strcuture used to store data as a key value pair.)
         // The difference between a hashmap and a map is that a hashmap is less flexible and uses a hashtable for fast access)( a hashtable is a collection of key-value pairs, for each value you have a unique key for it)
```

Comment:
`HashMap` is just an implementation of the `Map` interface. There are other implementations of the interface available in Java including the `LinkedHashMap` implementation.

https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html